mixnet/node.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import annotations
+
+import queue
+import threading
+import time
+from dataclasses import dataclass
+from threading import Thread
+from typing import Tuple, TypeAlias
+
+from cryptography.hazmat.primitives.asymmetric.x25519 import (
+    X25519PrivateKey,
+    X25519PublicKey,
+)
+from pysphinx.node import Node
+from pysphinx.sphinx import (
+    Payload,
+    ProcessedFinalHopPacket,
+    ProcessedForwardHopPacket,
+    SphinxPacket,
+    UnknownHeaderTypeError,
+)
+
+from mixnet.bls import BlsPrivateKey, BlsPublicKey
+from mixnet.poisson import poisson_interval_sec
+
+NodeId: TypeAlias = BlsPublicKey
+# 32-byte that represents an IP address and a port of a mix node.
+NodeAddress: TypeAlias = bytes
+
+PacketQueue: TypeAlias = "queue.Queue[Tuple[NodeAddress, SphinxPacket]]"
+PacketPayloadQueue: TypeAlias = (
+    "queue.Queue[Tuple[NodeAddress, SphinxPacket | Payload]]"
+)
+
+
+@dataclass
+class MixNode:
+    identity_private_key: BlsPrivateKey
+    encryption_private_key: X25519PrivateKey
+    addr: NodeAddress
+
+    def identity_public_key(self) -> BlsPublicKey:
+        return self.identity_private_key.get_g1()
+
+    def encryption_public_key(self) -> X25519PublicKey:
+        return self.encryption_private_key.public_key()
+
+    def sphinx_node(self) -> Node:
+        return Node(self.encryption_private_key, self.addr)
+
+    def start(
+        self,
+        delay_rate_per_min: int,
+        inbound_socket: PacketQueue,
+        outbound_socket: PacketPayloadQueue,
+    ) -> MixNodeRunner:
+        thread = MixNodeRunner(
+            self.encryption_private_key,
+            delay_rate_per_min,
+            inbound_socket,
+            outbound_socket,
+        )
+        thread.daemon = True
+        thread.start()
+        return thread
```

Comment:
Nitpick: could this be async instead?

Reply:
> I haven't used async code in this repo. But, hm, let me translate this into async. If it doesn't look complicated, we can go with async because it's much easier to approximate M/M/∞ queue using async rather than sync.

It is not necessary!, we can go ahead with this.

Reply:
Yeah, but after opening subsequent PRs #50 and #51, now I have a feeling that async code would be even simpler. I'll try it and open a separate PR soon. So, I hope this and subsequent PRs are reviewed and merged independently.